cl/alerts/views.py

```diff
@@ -161,3 +170,29 @@ def new_docket_alert(request: AuthenticatedHttpRequest) -> HttpResponse:
             "private": True,
         },
     )
+
+
+@ratelimit_deny_list
+def subscribe_docket_alert(request, secret_key):
+    docket_alert = get_object_or_404(DocketAlert, secret_key=secret_key)
+    docket_alert.alert_type = DocketAlert.SUBSCRIPTION
+    docket_alert.save()
+    return render(
+        request,
+        "enable_docket_alert.html",
+        {"docket_alert": docket_alert, "private": True},
+    )
+
+
+@ratelimit_deny_list
+def unsubscribe_docket_alert(request, secret_key):
+    docket_alert = get_object_or_404(DocketAlert, secret_key=secret_key)
+    docket_alert.alert_type = DocketAlert.UNSUBSCRIPTION
+    docket_alert.save()
+    # Send Unsubscription confirmation email to the user
+    send_unsubscription_confirmation.delay(docket_alert.pk)
+    return render(
+        request,
+        "disable_docket_alert.html",
+        {"docket_alert": docket_alert, "private": True},
+    )
```

Comment:
Sure! I've created `flip_docket_alert` function to encapsulate the duplicated code for subscribing and unsubscribe views,  however, I think it's not similar to `toggle_docket_alert` because this one checks if the `DocketAlert ` exists first and also receives the docket_id from AJAX, while `flip_docket_alert` get the DocketAlert from the secret key.